AlchemyData News widgets and endpoints

michaelkeeling opened the pull request:
* Updated the the AlchemyData News widgets to use app builder endpoints.
* Cleaned up the code a little.
* Migrating the code into the App Builder samples repo so it's easier to find.

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+# Integrating the Watson Developer Cloud AlchemyAPI News Service with Watson Explorer
```
There are many references to "Watson Developer Cloud", WDC, and Bluemix.  Most can be dropped if we are pointing right at the Alchemy service now.

michaelkeeling replied:
cleaned up a bit.  I included a few references to Bluemix since you can access Alchemy News through either Bluemix or the AlchemyAPI website.  The tutorials should be neutral now insofar as which approach is used.

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+## Prerequisites
+Please read "[Introduction to the Watson Developer Cloud for Application Builder Administrators](https://github.com/IBM-Watson/wex-appbuilder-samples/blob/master/intro-to-bluemix-for-app-builder.md)" for an overview of the integration architecture and information on tools and libraries for required to work with Bluemix.
+
+- An [IBM Bluemix](https://ace.ng.bluemix.net/) account
```
Bluemix is no longer a prereq

michaelkeeling replied:
optional prerequisites are now included.

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+
+1. Watson Explorer App Builder endpoints that will call out to AlchemyData News APIs.
+2. Watson Explorer App Builder widgets that use the created endpoints.
+  - The news article widget sends displays a list of top news articles with sentiment for a given keyword search.
```
"sends" "displays"

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+
+
+
+#### Configuring the News Trend Widget
```
It looks like there is a bit of mix up between the "list" and "trend" widgets here in the descriptions, and in the specific steps below.

michaelkeeling replied:
Thanks for catching that.  Not sure what happened there.

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+5. Copy and paste the [code for this widget](appbuilder/alchemy-news-list-widget.erb) into the Type-specific Configuration.
+6. Save the widget.
+7. Go back to the Book Title > detail page.
+8. Drag the `WDC_AlchemyAPI_News` widget to the top of the right column and save the page configuration.
```
WDC_AlchemyAPI_News: Is this the right name?

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+8. Drag the `WDC_AlchemyAPI_News` widget to the top of the right column and save the page configuration.
+
+
+#### Configuring the News List Widget
```
It looks like there is a bit of mix up between the "list" and "trend" widgets here in the descriptions, and in the specific steps below.

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+5. Copy and paste the [code for this widget](appbuilder/alchemy-news-trend-widget.erb) into the Type-specific Configuration.
+6. Save the widget.
+7. Go back to the Book Title > detail page.
+8. Drag the `WDC_AlchemyAPI_News` widget to the top of the right column and save the page configuration.
```
WDC_AlchemyAPI_News: Is this the right name?

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+At this point the widget should be fully configured.  However, you may find far more relevant news results for your data if you specify the entity type as part of the news query.  See the AlchemyAPI documentation for [details about using entity types](http://docs.alchemyapi.com/v1.0/docs/entities) as well as [nearly 400 other parameters](http://docs.alchemyapi.com/v1.0/docs/full-list-of-supported ...
+
+- **Improved Search Accuracy** - AlchemyData provides [nearly 400 other parameters](http://docs.alchemyapi.com/v1.0/docs/full-list-of-supported-news-api-fields) that can be used to construct sophisticated News API queries.  This sample demonstrates a simple starting query.
+- **Privacy and Security** - The AlchemyData News widget makes a web request to the Bluemix application endpoint configured in the Application Builder proxy.  In this example, this call is made over an unencrypted and unauthenticated HTTP connection, but your Bluemix application can be modified to support better security.
```
Sounds real Bluemixy, and we can use HTTPS now.

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+
+- **Improved Search Accuracy** - AlchemyData provides [nearly 400 other parameters](http://docs.alchemyapi.com/v1.0/docs/full-list-of-supported-news-api-fields) that can be used to construct sophisticated News API queries.  This sample demonstrates a simple starting query.
+- **Privacy and Security** - The AlchemyData News widget makes a web request to the Bluemix application endpoint configured in the Application Builder proxy.  In this example, this call is made over an unencrypted and unauthenticated HTTP connection, but your Bluemix application can be modified to support better security.
+- **Failures will happen** - All distributed systems are inherently unreliable and failures will inevitably occur when attempting to call out to Bluemix.  Consider how failures should be handled.
```
Instead of "Bluemix" maybe "Alchemy News service"

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+- **Privacy and Security** - The AlchemyData News widget makes a web request to the Bluemix application endpoint configured in the Application Builder proxy.  In this example, this call is made over an unencrypted and unauthenticated HTTP connection, but your Bluemix application can be modified to support better security.
+- **Failures will happen** - All distributed systems are inherently unreliable and failures will inevitably occur when attempting to call out to Bluemix.  Consider how failures should be handled.
+- **Data Preparation** - It is the responsibility of the caller to ensure that representative data is being sent to the AlchemyAPI News API.  Additional data preparation may be required in some cases.  Choose content carefully and be sure you are sending a clean news query.
+- **Scalability** - This example uses only a single cloud instance with the default Bluemix application settings.  In a production scenario consider how much hardware will be required and adjust the Bluemix application settings accordingly.
```
Bluemixy

eoehlsch commented on alchemy_news/appbuilder/alchemy-news-list-endpoint.rb:
```diff
@@ -0,0 +1,52 @@
+news_url = "http://access.alchemyapi.com/calls/data/GetNews"
```
I was going to suggest using HTTPS, but found that the docs now say https://gateway-a.watsonplatform.net/calls/data/GetNews

eoehlsch commented on alchemy_news/appbuilder/alchemy-news-trend-endpoint.rb:
```diff
@@ -0,0 +1,29 @@
+news_url = "http://access.alchemyapi.com/calls/data/GetNews"
```
I was going to suggest using HTTPS, but found that the docs now say https://gateway-a.watsonplatform.net/calls/data/GetNews

eoehlsch commented on alchemy_news/appbuilder/alchemy-news-trend-endpoint.rb:
```diff
@@ -0,0 +1,29 @@
+news_params = {
+  :apikey => 'YOUR_API_KEY_HERE',
+  :outputMode => 'json',
+  :count => 10
```
What does the 'count' parameter do here?  I honestly can't find an answer in the docs...

michaelkeeling commented:
@eoehlsch - I've addressed all your comments.  I plan to merge this EOD tomorrow unless you see any major show stoppers.  We can always fix any other problems with the sample later.

eoehlsch commented on alchemy_news/README.md:
```diff
@@ -0,0 +1,152 @@
+
+#### Configuring the News List Widget
+
+The purpose of this example widget is to perform a AlchemyData News API search for the entity name and display a list of recnet news articles.  The results of the AlchemyData News search are displayed in the Application Builder UI.
```
"recnet" should be "recent"
